doc/Hash/Policy-based-hash.md

```diff
@@ -0,0 +1,219 @@
+# Policy-based Hash
+
+Title       | Policy-based Hash
+------------|----------------
+Authors     | Nvidia
+Status      | In review
+Type        | Standards track
+Created     | 04/29/2020
+SAI-Version | 1.6
+
+Sai defines two metadata values for ECMP and LAG hash.
+The fields that are used for their calculation are defined at the switch level with attributes `SAI_SWITCH_ATTR_ECMP_HASH` and `SAI_SWITCH_ATTR_LAG_HASH`.
+There are also hash objects that can be assigned for some basic packet types.
+
+But if we would like to cover some use cases that are not defined in SAI, we need to be more flexible in a way that won't require changing the API.
+These use cases will be covered by a concept introduced here called policy-based hashing.
+
+```
+    /** Set custom LAG hash object ID */
+    SAI_ACL_ACTION_TYPE_SET_LAG_HASH_ID,
+
+    /** Set custom ECMP hash object ID */
+    SAI_ACL_ACTION_TYPE_SET_ECMP_HASH_ID,
+
+} sai_acl_action_type_t;
+```
+
+The two action types are not conflicting between one another, but the same action type used more than once for the parralel execution is.
+
+It utilizes the ACL infrastructure with the new actions to set custom hash object for a special ECMP or LAG treatment.
+
+Also, the hash object itself is extended to be able to specify the portions of the fields that we are interested in, like, for example, only some bytes from IPv6 address.
+
+```
+    /**
+     * @brief Hash fine-grained field list
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_FINE_GRAINED_HASH_FIELD
+     * @default empty
+     */
+    SAI_HASH_ATTR_FINE_GRAINED_HASH_FIELD_LIST,
+```
+
+Optional ordering of the fields is available as well.
+
+```
+    /**
+     * @brief Optional field ordering.
+     *
+     * Specifies in which order the fields are hashed,
+     * and defines in which fields should be associative for CRC with the same sequence ID.
+     * If not provided, it's up to SAI driver to choose.
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_ONLY
+     * @default 0
+     */
+    SAI_FINE_GRAINED_HASH_FIELD_ATTR_SEQUENCE_ID,
+```
+
+The example below provides a complete view of the capabilities of the policy-based hashing:
+
+```
+/*****************************************************
+ * Create fine grained fields for custom hash
+ *****************************************************/
+ 
+ sai_object_id_t inner_v6_hash_oid;
+
+ attr.id = SAI_FINE_GRAINED_HASH_FIELD_ATTR_NATIVE_HASH_FIELD;
+ attr.value.s32 = SAI_NATIVE_HASH_FIELD_INNER_IP_PROTOCOL;
+ attrs.push_back(attr);
+ 
+ attr.id = SAI_FINE_GRAINED_HASH_FIELD_ATTR_U8_MASK;
```

Comment:
where can I find this attribute?

Reply:
Thet's from the old proposal. Removed from md now.